src/compiler/checker.ts

```diff
@@ -35720,13 +35719,13 @@ namespace ts {
                 return;
             }
 
-            const testedSymbol = getSymbolAtLocation(testedNode);
-            if (!testedSymbol) {
+            const testedSymbol = testedNode && getSymbolAtLocation(testedNode);
+            if (!testedSymbol && !isPromise) {

```

Comment:
I don’t know that we actually need have divergent behavior for unawaited promises and uncalled expressions here—what do we expect for something like

```ts
const or = x => y => x || y;

if (or(true)) { // today: no error, but maybe should be?
  // ...
}
```